File: features-mobile/features-headlines/src/main/java/com/kashapovrush/headlines_features/di/ViewModelModule.java
package com.kashapovrush.headlines_features.di;

import androidx.lifecycle.ViewModel;

import com.example.features_common.di.ViewModuleKey;
import com.example.features_common.viewmodel.CommonViewModel;

import dagger.Binds;
import dagger.Module;
import dagger.multibindings.IntoMap;

@Module
interface ViewModelModule {
    @IntoMap
    @Binds
    @ViewModuleKey(CommonViewModel.class)
    ViewModel bindViewModel(CommonViewModel viewModel);
}

File: features-mobile/features-search/src/main/java/com/example/features_search/di/ViewModelModule.kt
package com.example.features_search.di

import androidx.lifecycle.ViewModel
import com.example.features_common.di.ViewModuleKey
import com.example.features_common.viewmodel.CommonViewModel
import dagger.Binds
import dagger.Module
import dagger.multibindings.IntoMap

@Module
interface ViewModelModule {

    @IntoMap
    @Binds
    @ViewModuleKey(CommonViewModel::class)
    fun bindViewModel(viewModel: CommonViewModel): ViewModel
}